mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulPuzzleSelectionScreen.kt

```diff
@@ -83,8 +83,9 @@ data class PuzzleInfoAdapter(
 fun Puzzle.toPuzzleInfoAdapter(): PuzzleInfoAdapter {
   val playerColor =
       when (this.boardSnapshot.playing) {
-        Color.White -> ChessBoardState.Color.White
-        Color.Black -> ChessBoardState.Color.Black
+        // Inverted colors to FEN since first UCI moves describes "computer" move
+        Color.White -> ChessBoardState.Color.Black
+        Color.Black -> ChessBoardState.Color.White
```

Comment:
You could replace this with `this.boardSnapshot.playing.other().toColor()`.

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/309/commits/246a052bf444aa34b9afb8521ff40291cfc8fc31